### ISSUE TITLE
2.1-rc3 java.lang.ClassCastException error when inserting the same value 40 times into nonunique index

### ISSUE THREAD

**Issue by flyingrobots** [orientechnologies/orientdb#4343]
1. Create a subclass of `V` called `Player` with a `STRING` property, `name`, and an `INTEGER` property, `power`. 
2. Create a `unique` index for `Player.name`.
2. Create a `nonunique` index for `Player.power`.
3. Attempt to create 100 `Player` vertices, each one with a unique name (like `Player1`, `Player2`, etc.), but make each record with the same `power` value: `123`. 

The first 39 vertices will be created successfully, however, any further attempts to create `Player` records will fail with the following error:

```
2015-06-10 17:44:32:222 SEVERE Sent run-time exception to the client /<IP ADDRESS>:<CONNECTION ID>: com.orientechnologies.orient.core.exception.OTransactionException: Error on committing distributed transaction [ONetworkProtocolBinary]Error on distributed transaction commit
com.orientechnologies.orient.core.index.OIndexException: Error during insertion of key in index
    at com.orientechnologies.orient.core.index.OIndexMultiValues.put(OIndexMultiValues.java:141)
    at com.orientechnologies.orient.core.index.OIndexMultiValues.put(OIndexMultiValues.java:52)
    at com.orientechnologies.orient.core.index.OIndexAbstract.putInSnapshot(OIndexAbstract.java:870)
    at com.orientechnologies.orient.core.index.OIndexAbstract.applyIndexTxEntry(OIndexAbstract.java:1050)
    at com.orientechnologies.orient.core.index.OIndexAbstract.addTxOperation(OIndexAbstract.java:684)
    at com.orientechnologies.orient.core.tx.OTransactionOptimistic$CommitIndexesCallback.run(OTransactionOptimistic.java:100)
    at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:930)
    at com.orientechnologies.orient.server.distributed.ODistributedStorage.commit(ODistributedStorage.java:882)
    at com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit(OTransactionOptimistic.java:590)
    at com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(OTransactionOptimistic.java:149)
    at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:2486)
    at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:2456)
    at com.orientechnologies.orient.server.distributed.task.OTxTask.execute(OTxTask.java:129)
    at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.executeOnLocalNode(OHazelcastPlugin.java:776)
    at com.orientechnologies.orient.server.hazelcast.ODistributedWorker.onMessage(ODistributedWorker.java:300)
    at com.orientechnologies.orient.server.hazelcast.ODistributedWorker.run(ODistributedWorker.java:121)
Caused by: java.lang.ClassCastException
```
NOTE the error is of the type `java.lang.ClassCastException` - not sure why this only starts to happen after the first 39 records have been created.

NOTE If you give the `Player` records unique power values, you will not encounter this error.

NOTE If you drop the `Player.power` index, you will be able to create as many `Player` vertices as you wish, as long as they have unique names, of course.

**Comment by tglman** [comment]
i wrote a relative test case but i was not able to reproduce it, can you give more details? are you sure that your classPath is clean? often ClassCastException come out because of dirty classpath

here is my test case:
```
  @Test
  public void testInsert40Entries() {
    OrientGraph db = new OrientGraph("memory:" + OCompositeIndexMultipleEntryTest.class.getSimpleName(), false);
    try {
      OrientVertexType type = db.createVertexType("Player");
      OProperty propName = type.createProperty("name", OType.STRING);
      propName.createIndex(INDEX_TYPE.UNIQUE);
      OProperty propPower = type.createProperty("power", OType.INTEGER);
      propPower.createIndex(INDEX_TYPE.NOTUNIQUE);
      db.begin();

      for (int i = 0; i < 100; i++) {
        Vertex v = db.addVertex(null);
        v.setProperty("name", "Player" + i);
        v.setProperty("power", 123);
      }
      db.commit();
    } finally {
      db.drop();
    }

  }
```

**Comment by lvca** [comment]
This seems an old issue already fixed in 2.1-rc4. Closing it. In case somebody has the same problem with 2.1-rc5 or later, please comment the issue.
